saltstack/base/salt/prerequisites/user_uid.sls

```diff
@@ -0,0 +1,34 @@
+{% set ids = salt['grains.filter_by']({
+    'amazon-ebs': {
+        'cloudera_scm_user': '992',
+        'cloudera_scm_group': '988',
+    },
+    'azure-arm': {
+        'cloudera_scm_user': '991',
+        'cloudera_scm_group': '987',
+    },
+    'googlecompute': {
+        'cloudera_scm_user': '992',
+        'cloudera_scm_group': '988',
+    },
+},
+grain='builder_type',
+default='amazon-ebs'
+)%}
+
+create_cloudera_scm_group:
+  group.present:
+    - name: cloudera-scm
+    - gid: {{ ids.cloudera_scm_group }} 
+
+create_cloudera_scm_user:
+  user.present:
+    - name: cloudera-scm
+    - fullname: Cloudera Manager
+    - shell: /sbin/nologin
+    - home: /var/lib/cloudera-scm-server
+    - createhome: False
+    - uid: {{ ids.cloudera_scm_user }}
+    - gid: {{ ids.cloudera_scm_group }}     
+    - groups:
+      - cloudera-scm
```

Comment:
I would add a sanity check to make sure the user and group exist regardless of the builder_type. That way we can be sure that it works correctly even if the type changes or we add a new type.

Reply:
Makes sense, but the dictionary has a default clause meaning all new builders will inherit that setting

Reply:
Right, I can merge without that check. Do you want me or should we wait for the Holidays to end?